File: credit.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main (void)
{
    // Ask for the card number.
    long number;
    do
    {
        number = get_long("Number: ");
    }
    while (number < (long) pow(10,12));
    
    int check;
    int front = 0;
    for (int i = 1; i <= 16; i += 1)
    {
        // Sum of the numbers multiplied by 2.
        if (i % 2 == 0)
        {
            int a = (number % (long) pow(10, i)) / (long) pow(10, i - 1);
            int b = 2 * a;
            if (b >= 10)
            {
                b = 1 + b % 10;
            }
            check = check + b;       
        }
        // Sum to the previous sum the sum of the other numbers.
        else if (i % 2 == 1)
        {
            int c = (number % (long) pow(10, i)) / (long) pow(10, i - 1);
            check = check + c; 
        }        
    }
 
    // Calculate of the two leading digits of the number. 
    int dec = 0;
    int i = 16;
    do
    {
        dec = number / (long) pow(10, i);
        if (dec != 0)
        {
            int unit = number % (long) pow(10, i) / (long) pow(10, i - 1);
            front = dec * 10 + unit; 
            break;         
        }
        i--;
    }
    while (dec == 0);

    // Perform the 10module of the checknum.
    check = check % 10;
    
    // Check if the module of is 0 and the characteristics of all different cards.
    if (check == 0 && number > pow(10, 14) && number < pow(10, 15) && (front == 34 || front == 37))
    {
            printf("AMEX\n");
    }
    else if (check == 0 && number > pow(10, 15) && number < pow(10, 16) && front >= 51 && front <= 55)
    {
        printf("MASTERCARD\n");
    }
    else if (check == 0  && number > pow(10, 12) && number < pow(10, 16) && front >= 40 && front < 50)
    {
        printf("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }  
}
    

File: credit.py
#!/usr/bin/env python3
from cs50 import get_int


def main():
    number = get_int("Number: ")
    check = 0
    front = 0
    for i in range(17):
        # Sum of the numbers multiplied by 2.
        if i % 2 == 0:
            a = (number % pow(10, i)) // pow(10, i - 1)
            b = 2 * a
            if b >= 10:
                b = 1 + b % 10
            check = check + b
        # Sum to the previous sum the sum of the other numbers.
        elif i % 2 == 1:
            c = (number % pow(10, i)) // pow(10, i - 1)
            check = check + c

    # Calculate of the two leading digits of the number.
    dec = 0
    i = 16
    while True:
        dec = number // pow(10, i)
        if dec != 0:
            unit = number % pow(10, i) // pow(10, i - 1)
            front = dec * 10 + unit
            break
        i -= 1
        if dec != 0:
            break

    # Perform the 10module of the checknum.
    check = check % 10

    # Check if the module of is 0 and the characteristics of all different cards.
    if check == 0 and number > pow(10, 14) and number < pow(10, 15) and (front == 34 or front == 37):
        print("AMEX")
    elif check == 0 and number > pow(10, 15) and number < pow(10, 16) and front >= 51 and front <= 55:
        print("MASTERCARD")
    elif check == 0 and number > pow(10, 12) and number < pow(10, 16) and front >= 40 and front < 50:
        print("VISA")
    else:
        print("INVALID")


if __name__ == "__main__":
    main()

# Fundamentally I' ve translated my own c code to python in this problem.
# https://github.com/me50/jcstark/blob/cs50/problems/2019/x/credit/credit.c